Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -388,6 +394,31 @@ public struct HTTPHeaders: CustomStringConvertible {
         }
         return headersArray.description
     }
+    
+    /// This is public to aid in the creation of supplemental HTTP libraries, e.g.
+    /// NIOHTTP2 and NIOHPACK. It is not intended for general use.
+    public static func createHeaderBlock(buffer: ByteBuffer, headers: [HTTPHeader]) -> HTTPHeaders {
+        return HTTPHeaders(buffer: buffer, headers: headers, keepAliveState: KeepAliveState.unknown)
+    }
+    
+    
+    /// Provides access to raw UTF-8 storage of the headers in this header block, along with
+    /// a list of the header strings' indices.
+    ///
+    /// - note: This is public to aid in the creation of supplemental HTTP libraries, e.g.
+    /// NIOHTTP2 and NIOHPACK. It is not intended for general use.
+    ///
+    /// - parameters:
+    ///   - block:      A block that will be provided UTF-8 header block information.
+    ///   - buf:        A raw `ByteBuffer` containing potentially-contiguous sequences of UTF-8 encoded
+    ///                 characters.
+    ///   - locations:  An array of `HTTPHeader`s, each of which contains information on the location in
+    ///                 the buffer of both a header's name and value.
+    ///   - contiguous: A `Bool` indicating whether the headers are stored contiguously, with no gaps between
+    ///                 adjacent name/value pairs.
```

Comment:
Do you mind elaborating on this last parameter? Specifically, there *are* gaps, but those gaps correspond to a HTTP/1.1-formatted header block (that is, the gaps are either `: ` or `\r\n` in the buffer).

Reply:
Argh, I'd forgotten in contained those. I was thinking of the `continuous` property that is marked `false` when something is removed from the buffer, i.e. when the buffer also contains strings that aren't intended to be included in the header list. In other words, whether the buffer can be directly copied.

In retrospect, it doesn't really match my intended use case (i.e. whether `HPACKHeaders` can just grab the buffer directly without caring about copying extra data—since the HPACK structure isn't supporting the same add/remove functionality and would like a smaller storage load). However, it still seems useful since it can tell receivers if the buffer is already a valid encoded HTTP/1 header block, ready for the wire as-is.